ql/src/java/org/apache/hadoop/hive/ql/exec/tez/AvailableSlotsCalculator.java

```diff
@@ -0,0 +1,32 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.hadoop.hive.ql.exec.tez;
+
+import org.apache.hadoop.conf.Configuration;
+import org.apache.tez.runtime.api.InputInitializerContext;
+
+/**
+ * Interface for fetching available slots during split calculation
+ */
+public interface AvailableSlotsCalculator {
+
+    void initialize(Configuration conf);
+
+    int getAvailaleSlots(InputInitializerContext inputInitializerContext);
```

Comment:
@adesh-rao : could you please describe this scenario and a possible implementation class further?
jira description implies that you want to get rid of tightly coupled tez input context, but the interface has tez InputInitializerContext in the signature, so I don't think we addressed the original description

Reply:
Removed the dependency of tez input context on the interface.